Source/NSButtonCell.m

```diff
@@ -1922,4 +1922,19 @@ - (id) initWithCoder: (NSCoder*)aDecoder
   return self;
 }
 
+// Implement 10.7 NSRadio button behavior...
+
+- (BOOL) _isRadio
+{
+  return ([self image] == [NSImage imageNamed: @"NSRadioButton"]);
+}
+
+- (NSInteger) nextState
```

Comment:
I don't understand this method, could you please give me a short explanation? This surely implement the "prevent switch to off if button is currently on and is a radio button" part of your commit message but what is it needed for?

Reply:
nextState changes the state from the previous state to the state it should be in after... NSOffState -> NSOnState, etc.   On 10.7+ if you click on a radio button that is on it remains on.   This change is to ensure that happens.

Reply:
Thank you for the explanation. Does this new behaviour happen for a single radio button as well? I understand how this makes sense for a group of radio buttons, but on a single it would prevent switching it off.

Reply:
I just tested it by removing one of the buttons from the test mentioned above.   You can also do this by adding another radio button connected to a different selector.   It does come on when clicked, but it does not go off when clicked again.

Reply:
@fredkiefer Let me know if this latest change looks good.